lidarrmetadata/chart.py

```diff
@@ -0,0 +1,171 @@
+    response = requests.get(URL)
+    results = response.json()['feed']['results']
+
+    search_provider = provider.get_providers_implementing(provider.AlbumNameSearchMixin)[0]
+
+    search_results = []
+    for result in results:
+        search_result = search_provider.search_album_name(result['name'], result['artistName'])
+        if search_result:
+            search_result = search_result[0]
+            search_results.append(_parse_album_search_result(search_result))
+
+            if len(search_results) == count:
+                break
+
+    return search_results
+
+
+def get_billboard_200_albums_chart(count=10):
+    """
+    Gets billboard top 200 albums
+    :param count: Number of results to return. Defaults to 10
+    :return: Chart response for billboard-200
+    """
+    results = billboard.ChartData('billboard-200')
+
+    search_provider = provider.get_providers_implementing(provider.AlbumNameSearchMixin)[0]
+
+    search_results = []
+    for result in results:
+        search_result = search_provider.search_album_name(result.title, result.artist)
+        if search_result:
+            search_result = search_result[0]
+            search_results.append(_parse_album_search_result(search_result))
+
+            if len(search_results) == count:
+                break
+
+    return search_results
+
+
+def get_itunes_chart(count=10):
+    """
+    Gets and parses itunes chart
+    :param count: Number of results to return. Defaults to 10
+    :return: Chart response for itunes
+    """
+    URL = 'https://rss.itunes.apple.com/api/v1/us/itunes-music/top-albums/all/{count}/explicit.json'.format(count=4 * count)
+    response = requests.get(URL)
+    results = response.json()['feed']['results']
+
+    search_provider = provider.get_providers_implementing(provider.AlbumNameSearchMixin)[0]
+
+    search_results = []
+    for result in results:
+        search_result = search_provider.search_album_name(result['name'], result['artistName'])
+        if search_result:
+            search_result = search_result[0]
+            search_results.append(_parse_album_search_result(search_result))
+
+            if len(search_results) == count:
+                break
+
+    return search_results
+
+
+def get_lastfm_album_chart(count=10, user=None):
+    """
+    Gets and parses lastfm chart
+    :param count: Number of results to return. Defaults to 10
+    :return: Parsed chart
+    """
+    client = pylast.LastFMNetwork(api_key=config.CONFIG.LASTFM_KEY, api_secret=config.CONFIG.LASTFM_SECRET)
+
+    if user:
+        user = client.get_user(user[0])
+        lastfm_albums = user.get_top_albums()
+    else:
+        tag = client.get_tag('all')
+        lastfm_albums = tag.get_top_albums()
+
+    album_provider = provider.get_providers_implementing(provider.AlbumByIdMixin)[0]
+    albums = []
+    for result in pylast.extract_items(lastfm_albums):
+        # TODO Figure out a cleaner way to do this
+        rgid = album_provider.map_query(
```

Comment:
Do we need all this or can we just return a mbid that last fm gives us

Reply:
Unfortunately it seems to be needed. Lastfm returns a null artist and artist id quite often and the mbid returned for an album is for the release instead of release group.